azure/services/managedclusters/client.go

```diff
@@ -78,11 +78,19 @@ func (ac *AzureClient) GetCredentials(ctx context.Context, resourceGroupName, na
 }
 
 // CreateOrUpdate creates or updates a managed cluster.
-func (ac *AzureClient) CreateOrUpdate(ctx context.Context, resourceGroupName, name string, cluster containerservice.ManagedCluster) (containerservice.ManagedCluster, error) {
+func (ac *AzureClient) CreateOrUpdate(ctx context.Context, resourceGroupName, name string, cluster containerservice.ManagedCluster, headers map[string]string) (containerservice.ManagedCluster, error) {
```

Comment:
this is another case where it won't work super well with the [async interface](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/services/async/interfaces.go#L48) once we move this func to async... @shysank @Jont828 @sonasingh46 

For now I think it's fine to add it in here as-is but we'll need to have a way to expand ResourceSpecGetter when custom fields are needed like here

Reply:
actually for this one particularly @jackfrancis pointed out it would be something useful to have for all clients, so we could just add it as a generic capability of ResourceSpecGetter (ie.` spec.CustomHeaders`). Not a comment for this PR, just thinking out loud.

Reply:
>For now I think it's fine to add it in here as-is but we'll need to have a way to expand ResourceSpecGetter when custom fields are needed like here

Right. Also, I had some ideas on this one that I pasted on slack sometime back. I will open up an issue/discussion -- highlighting the challenges that we faced and we can actually brainstorm/groom to get to some solution.